src/callbacks.c

```diff
@@ -1469,22 +1470,29 @@ void on_context_action1_activate(GtkMenuItem *menuitem, gpointer user_data)
 		!EMPTY(doc->file_type->context_action_cmd))
 	{
 		command = g_strdup(doc->file_type->context_action_cmd);
+		format = _("Cannot execute context action command \"%s\": %s. "
+			"Check the path setting in Preferences.");
 	}
 	else
 	{
 		command = g_strdup(tool_prefs.context_action_cmd);
+		format = _("Cannot execute context action command \"%s\": %s. "
+			"Check the path setting in Filetype configuration.");
 	}
 
 	/* substitute the wildcard %s and run the command if it is non empty */
 	if (G_LIKELY(!EMPTY(command)))
 	{
-		utils_str_replace_all(&command, "%s", word);
+		gchar *command_line = g_strdup(command);
 
-		if (!spawn_async(NULL, command, NULL, NULL, NULL, &error))
+		utils_str_replace_all(&command_line, "%s", word);
+
+		if (!spawn_async(NULL, command_line, NULL, NULL, NULL, &error))
 		{
-			ui_set_statusbar(TRUE, "Context action command failed: %s", error->message);
+			ui_set_statusbar(TRUE, format, command, error->message);
```

Comment:
Actually, it occurred to me a few days ago that not caring about %s is wrong, but I was  too busy setting up my new computer...

A %s with a selection containing any quotes may break the command parsing under *unix. Thus we'd better display the short version on G_SPAWN_ERROR, and the long version on G_SHELL_ERROR (== parse error). Of course, it may break the spawning too, if %s is at the start of the command, but that's a bit extreme.